tests/build.rs

```diff
@@ -102,6 +102,31 @@ src[..]Cargo.toml:1:5-1:6 expected a value\n\n"))
 }
 
 #[test]
+fn cargo_compile_duplicate_build_targets() {
+    let p = project("foo")
+        .file("Cargo.toml", r#"
+            [package]
+            name = "foo"
+            version = "0.0.1"
+            authors = []
+
+            [lib]
+            name = "main"
+            crate-type = ["dylib"]
+
+            [dependencies]
+        "#)
+        .file("src/main.rs", r#"
+            fn main() {}
+        "#);
+
+    assert_that(p.cargo_process("build"),
+                execs()
+                .with_stderr_contains("duplicate build target found:")
```

Comment:
Could you change this to match the exit code (101) as well as stderr exhaustively?